src/mon/MonitorDBStore.h

```diff
@@ -507,16 +507,12 @@ class MonitorDBStore
   }
 
   int get(const string& prefix, const string& key, bufferlist& bl) {
-    set<string> k;
-    k.insert(key);
-    map<string,bufferlist> out;
-
-    db->get(prefix, k, &out);
-    if (out.empty())
-      return -ENOENT;
-    bl.append(out[key]);
-
-    return 0;
+    bufferlist outbl;
+    int r = db->get(prefix, key, &outbl);
+    if (!r) {
+      bl.append(outbl);
+    }
```

Comment:
I actually had different idea. Instead of that, it would be possible to add something like value_as_ptr() to KeyValueDB iterators, which return bufferptr instead of entire bufferlist. That way, db->get would be appending returned bufferptr to specified bufferlist, removing the need for two interim bufferlists (one in KeyValueDBIter->value() and another one in MonitorDBStore). Not sure if this should go with this PR though...